src/dbg/dbg.c

```diff
@@ -151,27 +151,24 @@ static void dbg_vprintf(int level, const char *fmt, va_list ap)
 static void dbg_fmt_vprintf(int level, const char *fmt, va_list ap)
 {
 	char buf[256];
-	int len;
 
 	dbg_lock();
+	int dbg_level   = dbg.level;
+	dbg_print_h *ph = dbg.ph;
+	void *arg       = dbg.arg;
+	dbg_unlock();
```

Comment:
Would does this locking protect?

Reply:
Should there be a lock in `dbg_handler_set()`?

Reply:
The mutex should protect all members of the struct.